orchagent/mirrororch.cpp

```diff
@@ -0,0 +1,745 @@
+
+    session.neighborInfo.resolved = false;
+    session.neighborInfo.neighbor = neighbor;
+    session.neighborInfo.mac = mac;
+
+    // Get port operation should not fail;
+    if (!m_portsOrch->getPort(session.neighborInfo.neighbor.alias, session.neighborInfo.port))
+    {
+        SWSS_LOG_ERROR("Failed to get port for %s alias\n", session.neighborInfo.neighbor.alias.c_str());
+        assert(false);
+    }
+
+    if (session.neighborInfo.port.m_type == Port::VLAN)
+    {
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_vlan_id;
+
+        Port member;
+        if (!m_fdbOrch->getPort(session.neighborInfo.mac, session.neighborInfo.vlanId, member))
+        {
+            return false;
+        }
+
+        session.neighborInfo.portId = member.m_port_id;
+        session.neighborInfo.resolved = true;
+
+        return true;
+    }
+    else if (session.neighborInfo.port.m_type == Port::LAG)
+    {
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_port_vlan_id;
+
+        if (session.neighborInfo.port.m_members.empty())
+        {
+            return false;
+        }
+
+        Port lagMember;
+        if (!m_portsOrch->getPort(*session.neighborInfo.port.m_members.begin(), lagMember))
+        {
+            SWSS_LOG_ERROR("Failed to get port for %s alias\n", session.neighborInfo.port.m_members.begin()->c_str());
+            assert(false);
+        }
+
+        session.neighborInfo.portId = lagMember.m_port_id;
+        session.neighborInfo.resolved = true;
+    }
+    else
+    {
+        session.neighborInfo.portId = session.neighborInfo.port.m_port_id;
+        session.neighborInfo.vlanId = session.neighborInfo.port.m_port_vlan_id;
+        session.neighborInfo.resolved = true;
+    }
+
+    return true;
+}
+
+bool MirrorOrch::activateSession(const string& name, MirrorEntry& session)
+{
+    sai_status_t status = SAI_STATUS_SUCCESS;
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+
+    SWSS_LOG_INFO("Activating mirror session %s\n", name.c_str());
+
+    assert(!session.status);
+
+    attr.id = SAI_MIRROR_SESSION_ATTR_TC;
+    attr.value.u8 = session.queue;
+    attrs.push_back(attr);
+
+    attr.id =  SAI_MIRROR_SESSION_ATTR_MONITOR_PORT;
+    attr.value.oid = session.neighborInfo.portId;
+    attrs.push_back(attr);
+
+    attr.id = SAI_MIRROR_SESSION_ATTR_TYPE;
+    attr.value.s32 = SAI_MIRROR_TYPE_ENHANCED_REMOTE;
+    attrs.push_back(attr);
+
+    attr.id =SAI_MIRROR_SESSION_ATTR_VLAN_TPID;
+    attr.value.u16 = ETH_P_8021Q;
+    attrs.push_back(attr);
+
+    attr.id =SAI_MIRROR_SESSION_ATTR_VLAN_ID;
+    attr.value.u16 = session.neighborInfo.vlanId;
+    attrs.push_back(attr);
+
+    attr.id = SAI_MIRROR_SESSION_ATTR_VLAN_PRI;
+    attr.value.u8 = MIRROR_SESSION_DEFAULT_VLAN_PRI;
+    attrs.push_back(attr);
+
+    attr.id = SAI_MIRROR_SESSION_ATTR_VLAN_CFI;
+    attr.value.u8 = MIRROR_SESSION_DEFAULT_VLAN_CFI;
+    attrs.push_back(attr);
```

Comment:
It is mandatory on  create:
https://github.com/opencomputeproject/SAI/blob/master/inc/saimirror.h#L136